### PR TITLE
Add Shipping Modules' Documentation

### PR DESCRIPTION
**Pull request by lat9**
Using the headings currently present in the wiki, I've filled in a couple of the blanks!

### CODE REVIEW

File: Developer_Documentation/1.6/code_docs/modules/shipping_modules.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+### Class Names
+A shipping method's filename (e.g. `myshipping`.php) is also used as its class name:
+```php
+class myshipping extends base
+{
+  ...
+}
+```
+**Note:** Since an order's shipping selection is stored as a the underscore (_) delimited combination of a shipping module's name and the shipping-method selected, a shipping module's `name` **cannot** include an underscore!
```
Comment (drbyte) [inline Developer_Documentation/1.6/code_docs/modules/shipping_modules.md]: `is stored as a the` feels ambiguous. 

I suggest rewording to:
```diff
-**Note:** Since an order's shipping selection is stored as a the underscore (_) delimited combination of a shipping module's name and the shipping-method selected, a shipping module's `name` **cannot** include an underscore!
+**Note:** Since an order's shipping selection is stored as a string combining a shipping module's name plus an underscore (_) plus the shipping-method selected, a shipping module's `name` **cannot** include an underscore!
```

**Comment by drbyte** [comment]
Thanks for submitting this!

File: Developer_Documentation/1.6/code_docs/modules/shipping_modules.md (hunk 1 above)
Comment (lat9) [inline Developer_Documentation/1.6/code_docs/modules/shipping_modules.md]: Sounds good, DrByte!  I wasn't happy with that wording either.

**Comment by drbyte** [comment]
👍

**Comment by zcwilt** [comment]
:+1:
